pkg/registries/registry.go

```diff
@@ -176,32 +178,52 @@ func (r Registry) RegistryName() string {
 }
 
 // NewRegistry - Create a new registry from the registry config.
-func NewRegistry(config Config, log *logging.Logger) (Registry, error) {
+func NewRegistry(con *config.Config, log *logging.Logger) (Registry, error) {
+	log.Debugf("configu - %#v", con)
 	var adapter adapters.Adapter
+	configuration := Config{
+		URL:        con.GetString("url"),
+		User:       con.GetString("user"),
+		Pass:       con.GetString("pass"),
+		Org:        con.GetString("org"),
+		Tag:        con.GetString("tag"),
+		Type:       con.GetString("type"),
+		Name:       con.GetString("name"),
+		Images:     con.GetSliceOfStrings("images"),
+		Namespaces: con.GetSliceOfStrings("namespaces"),
+		Fail:       con.GetBool("fail_on_error"),
+		WhiteList:  con.GetSliceOfStrings("white_list"),
+		BlackList:  con.GetSliceOfStrings("black_list"),
+		AuthType:   con.GetString("auth_type"),
+		AuthName:   con.GetString("auth_name"),
```

Comment:
Is this the thing you mentioned to me where we can have the config struct which gives me compile time validation while using the config?

Reply:
@jmrodri yes, another example is the broker config in the `AnsibleBroker` [struct](https://github.com/openshift/ansible-service-broker/pull/567/files#diff-55aca6fb74ac953881309c295ee9441eR127).

I think if we move to an Interface & Struct to the APB package, then this is a pattern that we can follow there.